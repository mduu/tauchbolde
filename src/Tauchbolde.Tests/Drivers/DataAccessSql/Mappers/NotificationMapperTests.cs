using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using Newtonsoft.Json;
using Tauchbolde.Domain.Types;
using Tauchbolde.Driver.DataAccessSql.DataEntities;
using Tauchbolde.Driver.DataAccessSql.Mappers;
using Xunit;

namespace Tauchbolde.Tests.Drivers.DataAccessSql.Mappers
{
    [UseReporter(typeof(DiffReporter))]
    public class NotificationMapperTests
    {
        [Fact]
        public void Map()
        {
            var dataObject = new NotificationData
            {
                Id = new Guid("674ED914-ACE3-4FF0-AD5C-4335CCF92607"),
                Message = "The Message",
               RecipientId = new Guid("2223CDFF-E5FE-4A68-81CD-BCC8800C365E"),
               Type = NotificationType.Neutral,
               EventId = new Guid("86E2735D-A537-43AD-9048-B1899A2E3F40"),
               AlreadySent = true,
               OccuredAt = new DateTime(2019, 8, 1),
               CountOfTries = 2,
               LogbookEntryId = new Guid("9BECCC85-BF62-4CAA-916E-3FB911868AE7"),
            };

            var domainObject = dataObject.MapTo();
            var newDataObject = domainObject.MapTo();
            
            newDataObject.Should().BeEquivalentTo(dataObject);
            Approvals.VerifyJson(JsonConvert.SerializeObject(
                new
                {
                    dataObject,
                    domainObject,
                    newDataObject
                }));
        }
    }
}