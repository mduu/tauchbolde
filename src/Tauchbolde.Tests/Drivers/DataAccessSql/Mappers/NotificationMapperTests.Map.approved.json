{
    "dataObject": {
        "RecipientId": "2223cdff-e5fe-4a68-81cd-bcc8800c365e",
        "Recipient": null,
        "OccuredAt": "2019-08-01T00:00:00",
        "AlreadySent": true,
        "CountOfTries": 2,
        "Message": "The Message",
        "Type": 3,
        "EventId": "86e2735d-a537-43ad-9048-b1899a2e3f40",
        "Event": null,
        "LogbookEntryId": "9beccc85-bf62-4caa-916e-3fb911868ae7",
        "LogbookEntry": null,
        "Id": "674ed914-ace3-4ff0-ad5c-4335ccf92607"
    },
    "domainObject": {
        "RecipientId": "2223cdff-e5fe-4a68-81cd-bcc8800c365e",
        "Recipient": null,
        "OccuredAt": "2019-08-01T00:00:00",
        "AlreadySent": true,
        "CountOfTries": 2,
        "Message": "The Message",
        "Type": 3,
        "EventId": "86e2735d-a537-43ad-9048-b1899a2e3f40",
        "Event": null,
        "LogbookEntryId": "9beccc85-bf62-4caa-916e-3fb911868ae7",
        "LogbookEntry": null,
        "Id": "674ed914-ace3-4ff0-ad5c-4335ccf92607"
    },
    "newDataObject": {
        "RecipientId": "2223cdff-e5fe-4a68-81cd-bcc8800c365e",
        "Recipient": null,
        "OccuredAt": "2019-08-01T00:00:00",
        "AlreadySent": true,
        "CountOfTries": 2,
        "Message": "The Message",
        "Type": 3,
        "EventId": "86e2735d-a537-43ad-9048-b1899a2e3f40",
        "Event": null,
        "LogbookEntryId": "9beccc85-bf62-4caa-916e-3fb911868ae7",
        "LogbookEntry": null,
        "Id": "674ed914-ace3-4ff0-ad5c-4335ccf92607"
    }
}